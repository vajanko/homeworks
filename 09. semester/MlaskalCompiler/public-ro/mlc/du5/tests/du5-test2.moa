Integer constants
0:	1
1:	729
2:	2
3:	3
4:	4
5:	5
6:	6
7:	30
8:	11
9:	10

Real constants

String constants
0:	"ahoj"
1:	" lidi"

Identifiers
0:	"BOOLEAN"
1:	"INTEGER"
2:	"REAL"
3:	"STRING"
4:	"FALSE"
5:	"TRUE"
6:	"READI"
7:	"READR"
8:	"READS"
9:	"P"
10:	"WRITEI"
11:	"WRITER"
12:	"WRITES"
13:	"WRITELN"
14:	"DU5TEST4"
15:	"DVA"
16:	"ARR"
17:	"CISLO"
18:	"S"
19:	"POLE"
20:	"MULTIPLY"
21:	"I"
22:	"J"
23:	"A"
24:	"B"
25:	"JEDNA"
26:	"PLUSJEDNA"
27:	"VYPISDVAKRAT"
28:	"R"

Functions and procedures
READI:INTEGER	[0;2], used registers=0
READR:REAL	[3;5], used registers=0
READS:STRING	[6;8], used registers=0
WRITEI(P:INTEGER [size=1;stkoffs=-1])	[9;11], used registers=0
WRITER(P:REAL [size=1;stkoffs=-1])	[12;14], used registers=0
WRITES(P:STRING [size=1;stkoffs=-1])	[15;17], used registers=0
WRITELN	[18;19], used registers=0
MULTIPLY(I:INTEGER [size=1;stkoffs=-2], J:INTEGER [size=1;stkoffs=-1]):INTEGER	[20;60], used registers=0
	VAR
		A:INTEGER[start=61;end61size=4;stkoffs=1]
		B:INTEGER[start=61;end61size=1;stkoffs=5]
JEDNA:INTEGER	[61;63], used registers=0
PLUSJEDNA(I:INTEGER [size=1;stkoffs=-1]):INTEGER	[64;68], used registers=0
VYPISDVAKRAT(I:INTEGER [size=1;stkoffs=-1])	[69;79], used registers=0
	VAR
		R:REAL[start=80;end80size=1;stkoffs=1]
DU5TEST4	[80;235], used registers=0
	VAR
		CISLO:INTEGER[start=236;end236size=1;stkoffs=0]
		S:STRING[start=236;end236size=1;stkoffs=1]
		POLE:INTEGER[start=236;end236size=4;stkoffs=2]

Instructions
READI:
0:	FNCREADI
1:	LSTI	[-1]
2:	RET
READR:
3:	FNCREADR
4:	LSTR	[-1]
5:	RET
READS:
6:	FNCREADS
7:	LSTS	[-1]
8:	RET
WRITEI:
9:	LLDI	[-1]
10:	FNCWRITEI
11:	RET
WRITER:
12:	LLDR	[-1]
13:	FNCWRITER
14:	RET
WRITES:
15:	LLDS	[-1]
16:	FNCWRITES
17:	RET
WRITELN:
18:	FNCWRITELN
19:	RET
MULTIPLY:
20:	INITI
21:	INITI
22:	INITI
23:	INITI
24:	INITI
25:	LDLITI	LITINT[3]
26:	LSTI	[5]
27:	LLDI	[-2]
28:	LLDI	[-1]
29:	MULI
30:	LREF	[1]
31:	LLDI	[5]
32:	LDLITI	LITINT[3]
33:	SUBI
34:	LDLITI	LITINT[2]
35:	MULI
36:	ADDP
37:	LDLITI	LITINT[2]
38:	LDLITI	LITINT[0]
39:	SUBI
40:	ADDP
41:	XSTI
42:	LREF	[1]
43:	LDLITI	LITINT[3]
44:	LDLITI	LITINT[3]
45:	SUBI
46:	LDLITI	LITINT[2]
47:	MULI
48:	ADDP
49:	LDLITI	LITINT[2]
50:	LDLITI	LITINT[0]
51:	SUBI
52:	ADDP
53:	XLDI
54:	LSTI	[-3]
55:	DTORI
56:	DTORI
57:	DTORI
58:	DTORI
59:	DTORI
60:	RET
JEDNA:
61:	LDLITI	LITINT[0]
62:	LSTI	[-1]
63:	RET
PLUSJEDNA:
64:	LLDI	[-1]
65:	LDLITI	LITINT[0]
66:	ADDI
67:	LSTI	[-2]
68:	RET
VYPISDVAKRAT:
69:	INITR
70:	LLDI	[-1]
71:	LDLITI	LITINT[2]
72:	MULI
73:	CVRTIR
74:	LSTR	[1]
75:	LLDR	[1]
76:	CALL	12	;; label=WRITER
77:	DTORR
78:	DTORR
79:	RET
DU5TEST4:
80:	INITI
81:	INITS
82:	INITI
83:	INITI
84:	INITI
85:	INITI
86:	INITI
87:	LDLITI	LITINT[5]
88:	LDLITI	LITINT[6]
89:	CALL	20	;; label=MULTIPLY
90:	DTORI
91:	DTORI
92:	GSTI	[0]
93:	LDLITI	LITINT[7]
94:	CVRTIR
95:	LDLITI	LITINT[8]
96:	CVRTIR
97:	DIVR
98:	CVRTRI
99:	GREF	[2]
100:	LDLITI	LITINT[0]
101:	LDLITI	LITINT[0]
102:	SUBI
103:	LDLITI	LITINT[2]
104:	MULI
105:	ADDP
106:	LDLITI	LITINT[3]
107:	LDLITI	LITINT[3]
108:	SUBI
109:	ADDP
110:	XSTI
111:	LDLITI	LITINT[7]
112:	LDLITI	LITINT[8]
113:	MODI
114:	GREF	[2]
115:	LDLITI	LITINT[0]
116:	LDLITI	LITINT[0]
117:	SUBI
118:	LDLITI	LITINT[2]
119:	MULI
120:	ADDP
121:	LDLITI	LITINT[4]
122:	LDLITI	LITINT[3]
123:	SUBI
124:	ADDP
125:	XSTI
126:	GREF	[2]
127:	LDLITI	LITINT[0]
128:	LDLITI	LITINT[0]
129:	SUBI
130:	LDLITI	LITINT[2]
131:	MULI
132:	ADDP
133:	LDLITI	LITINT[3]
134:	LDLITI	LITINT[3]
135:	SUBI
136:	ADDP
137:	XLDI
138:	LDLITI	LITINT[2]
139:	MULI
140:	GREF	[2]
141:	LDLITI	LITINT[2]
142:	LDLITI	LITINT[0]
143:	SUBI
144:	LDLITI	LITINT[2]
145:	MULI
146:	ADDP
147:	LDLITI	LITINT[3]
148:	LDLITI	LITINT[3]
149:	SUBI
150:	ADDP
151:	XSTI
152:	LDLITI	LITINT[0]
153:	GSTI	[0]
154:	GREF	[2]
155:	GLDI	[0]
156:	LDLITI	LITINT[0]
157:	SUBI
158:	LDLITI	LITINT[2]
159:	MULI
160:	ADDP
161:	LDLITI	LITINT[3]
162:	LDLITI	LITINT[3]
163:	SUBI
164:	ADDP
165:	XLDI
166:	CALL	9	;; label=WRITEI
167:	DTORI
168:	GREF	[2]
169:	LDLITI	LITINT[0]
170:	LDLITI	LITINT[0]
171:	SUBI
172:	LDLITI	LITINT[2]
173:	MULI
174:	ADDP
175:	LDLITI	LITINT[4]
176:	LDLITI	LITINT[3]
177:	SUBI
178:	ADDP
179:	XLDI
180:	LDLITI	LITINT[2]
181:	MULI
182:	GREF	[2]
183:	LDLITI	LITINT[2]
184:	LDLITI	LITINT[0]
185:	SUBI
186:	LDLITI	LITINT[2]
187:	MULI
188:	ADDP
189:	LDLITI	LITINT[4]
190:	LDLITI	LITINT[3]
191:	SUBI
192:	ADDP
193:	XSTI
194:	LDLITI	LITINT[9]
195:	GSTI	[0]
196:	GLDI	[0]
197:	CALL	69	;; label=VYPISDVAKRAT
198:	DTORI
199:	LDLITI	LITINT[5]
200:	LDLITI	LITINT[2]
201:	MULI
202:	LDLITI	LITINT[9]
203:	LDLITI	LITINT[4]
204:	LDLITI	LITINT[0]
205:	SUBI
206:	MODI
207:	ADDI
208:	GSTI	[0]
209:	LDLITS	LITSTR[0]
210:	GSTS	[1]
211:	GLDS	[1]
212:	LDLITS	LITSTR[1]
213:	ADDS
214:	GSTS	[1]
215:	GLDS	[1]
216:	CALL	15	;; label=WRITES
217:	DTORS
218:	INITI
219:	CALL	61	;; label=JEDNA
220:	CALL	9	;; label=WRITEI
221:	DTORI
222:	INITI
223:	LDLITI	LITINT[2]
224:	CALL	64	;; label=PLUSJEDNA
225:	DTORI
226:	CALL	9	;; label=WRITEI
227:	DTORI
228:	CALL	18	;; label=WRITELN
229:	DTORI
230:	DTORI
231:	DTORI
232:	DTORI
233:	DTORS
234:	DTORI
235:	HALT

