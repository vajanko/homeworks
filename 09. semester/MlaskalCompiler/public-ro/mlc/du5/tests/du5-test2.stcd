<?xml version="1.0" encoding="UTF-8"?>
<symbol_tables>
	<types>
		<RANGE type="T0" lower_bound="1" upper_bound="2"/>
		<RANGE type="T1" lower_bound="3" upper_bound="4"/>
		<INTEGER type="T2"/>
		<ARRAY type="T3" index_type="T1" element_type="T2"/>
		<ARRAY type="T4" index_type="T0" element_type="T3"/>
		<BOOLEAN type="T5"/>
		<RANGE type="T6" lower_bound="3" upper_bound="4"/>
		<RANGE type="T7" lower_bound="1" upper_bound="2"/>
		<ARRAY type="T8" index_type="T7" element_type="T2"/>
		<ARRAY type="T9" index_type="T6" element_type="T8"/>
		<REAL type="T10"/>
		<STRING type="T11"/>
	</types>
	<labels>
		<LABEL name="729"/>
	</labels>
	<symbols>
		<SKIND_TYPE name="ARR" type="T4"/>
		<SKIND_TYPE name="BOOLEAN" type="T5"/>
		<SKIND_GLOBAL_VARIABLE name="CISLO" type="T2"/>
		<SKIND_CONST name="DVA" value="2" type="T2"/>
		<SKIND_CONST name="FALSE" value="0" type="T5"/>
		<SKIND_TYPE name="INTEGER" type="T2"/>
		<SKIND_FUNCTION name="JEDNA" type="T2">
			<parameters>
			</parameters>
			<labels>
			</labels>
			<symbols>
			</symbols>
		</SKIND_FUNCTION>
		<SKIND_FUNCTION name="MULTIPLY" type="T2">
			<parameters>
				<PMODE_BY_VALUE name="I" type="T2"/>
				<PMODE_BY_VALUE name="J" type="T2"/>
			</parameters>
			<labels>
				<LABEL name="729"/>
			</labels>
			<symbols>
				<SKIND_LOCAL_VARIABLE name="A" type="T9"/>
				<SKIND_LOCAL_VARIABLE name="B" type="T2"/>
				<SKIND_LOCAL_VARIABLE name="I" type="T2"/>
				<SKIND_LOCAL_VARIABLE name="J" type="T2"/>
			</symbols>
		</SKIND_FUNCTION>
		<SKIND_FUNCTION name="PLUSJEDNA" type="T2">
			<parameters>
				<PMODE_BY_VALUE name="I" type="T2"/>
			</parameters>
			<labels>
			</labels>
			<symbols>
				<SKIND_LOCAL_VARIABLE name="I" type="T2"/>
			</symbols>
		</SKIND_FUNCTION>
		<SKIND_GLOBAL_VARIABLE name="POLE" type="T4"/>
		<SKIND_FUNCTION name="READI" type="T2">
			<parameters>
			</parameters>
			<labels>
			</labels>
			<symbols>
			</symbols>
		</SKIND_FUNCTION>
		<SKIND_FUNCTION name="READR" type="T10">
			<parameters>
			</parameters>
			<labels>
			</labels>
			<symbols>
			</symbols>
		</SKIND_FUNCTION>
		<SKIND_FUNCTION name="READS" type="T11">
			<parameters>
			</parameters>
			<labels>
			</labels>
			<symbols>
			</symbols>
		</SKIND_FUNCTION>
		<SKIND_TYPE name="REAL" type="T10"/>
		<SKIND_GLOBAL_VARIABLE name="S" type="T11"/>
		<SKIND_TYPE name="STRING" type="T11"/>
		<SKIND_CONST name="TRUE" value="1" type="T5"/>
		<SKIND_PROCEDURE name="VYPISDVAKRAT">
			<parameters>
				<PMODE_BY_VALUE name="I" type="T2"/>
			</parameters>
			<labels>
			</labels>
			<symbols>
				<SKIND_LOCAL_VARIABLE name="I" type="T2"/>
				<SKIND_LOCAL_VARIABLE name="R" type="T10"/>
			</symbols>
		</SKIND_PROCEDURE>
		<SKIND_PROCEDURE name="WRITEI">
			<parameters>
				<PMODE_BY_VALUE name="P" type="T2"/>
			</parameters>
			<labels>
			</labels>
			<symbols>
			</symbols>
		</SKIND_PROCEDURE>
		<SKIND_PROCEDURE name="WRITELN">
			<parameters>
			</parameters>
			<labels>
			</labels>
			<symbols>
			</symbols>
		</SKIND_PROCEDURE>
		<SKIND_PROCEDURE name="WRITER">
			<parameters>
				<PMODE_BY_VALUE name="P" type="T10"/>
			</parameters>
			<labels>
			</labels>
			<symbols>
			</symbols>
		</SKIND_PROCEDURE>
		<SKIND_PROCEDURE name="WRITES">
			<parameters>
				<PMODE_BY_VALUE name="P" type="T11"/>
			</parameters>
			<labels>
			</labels>
			<symbols>
			</symbols>
		</SKIND_PROCEDURE>
	</symbols>
</symbol_tables>
