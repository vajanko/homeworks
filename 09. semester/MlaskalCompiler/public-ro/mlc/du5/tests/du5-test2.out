Warning (line 46): Conversion loses significancy
Warning (line 2): Label '729' declared but never used
Warning (line 15): Label '729' declared but never used
Everything is nearly OK, although there were 3 warnings
